src/main/java/de/hysky/skyblocker/skyblock/speedPreset/SpeedPresets.java

```diff
@@ -0,0 +1,147 @@
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.Set;
+import java.util.function.BiConsumer;
+import java.util.regex.Pattern;
+
+public class SpeedPresets {
+
+	private static final Pattern COMMAND_PATTERN = Pattern.compile("^setmaxspeed\\s([a-zA-Z][a-zA-Z0-9_]*)$");
+
+	private static final Logger LOGGER = LoggerFactory.getLogger(SpeedPresets.class);
+	private static final Codec<Map<String, Short>> MAP_CODEC = Codec.unboundedMap(Codec.STRING, Codec.SHORT);
+	private static final File PRESETS_FILE = new File(SkyblockerMod.CONFIG_DIR.toFile(), "speed_presets.json");
+
+	private static SpeedPresets instance;
+
+	private final Map<String, Short> presets;
+
+	private SpeedPresets() {
+		this.presets = new LinkedHashMap<>();
+		this.loadPresets();
+	}
+
+	public static SpeedPresets getInstance() {
+		return instance == null ? instance = new SpeedPresets() : instance;
+	}
+
+	public static CommandNode<FabricClientCommandSource> getCommandNode() {
+		return ClientCommandManager.literal("setmaxspeed")
+				.requires(client -> Utils.isOnSkyblock())
+				.then(ClientCommandManager.argument("preset", StringArgumentType.string())
+						.suggests((ctx, builder) -> {
+							if (SkyblockerConfigManager.get().general.speedPresets.enableSpeedPresets) {
+								return CommandSource.suggestMatching(getInstance().presets.keySet(), builder);
+							}
+							return builder.buildFuture();
+						})).build();
+	}
+
+	@Init
+	public static void init() {
+		ClientSendMessageEvents.MODIFY_COMMAND.register((command) -> {
+			var matcher = COMMAND_PATTERN.matcher(command);
+			if (matcher.matches() && SkyblockerConfigManager.get().general.speedPresets.enableSpeedPresets) {
+				var presets = getInstance();
+				var preset = matcher.group(1);
+				if (presets.presets.containsKey(preset)) {
+					return String.format("setmaxspeed %d", presets.getPreset(preset));
+				}
+			}
+			return command;
+		});
+	}
+
+	public void clear() {
+		this.presets.clear();
+	}
+
+	public Set<Map.Entry<String, Short>> entries() {
+		return this.presets.entrySet();
+	}
+
+	public boolean hasPreset(String name) {
+		return this.presets.containsKey(name);
+	}
+
+	public short getPreset(String name) {
+		return this.presets.getOrDefault(name, (short) 0);
+	}
+
+	public void setPreset(String name, short value) {
+		this.presets.put(name, value);
+		savePresets();
+	}
+
+	public void forEach(BiConsumer<String, Short> consumer) {
+		this.presets.forEach(consumer);
+	}
+
+	public int getPresetCount() {
+		return this.presets.size();
+	}
+
+	public void loadPresets() {
+		try (var reader = Files.newReader(PRESETS_FILE, StandardCharsets.UTF_8)) {
+			var element = JsonParser.parseReader(reader);
+			var map = MAP_CODEC.decode(JsonOps.INSTANCE, element).resultOrPartial(LOGGER::warn);
+			map.ifPresent(mapResult -> this.presets.putAll(mapResult.getFirst()));
```

Comment:
```suggestion
			MAP_CODEC.parse(JsonOps.INSTANCE, element).resultOrPartial(LOGGER::error).ifPresent(this.presets::putAll);
```

Reply:
Would that even work? The resultorPartial now returns a Pair of A and T, in this case a pair of the Map and the JsonElement